prov/rxm/src/rxm_cq.c

```diff
@@ -737,6 +737,89 @@ static inline void rxm_cq_repost_rx_buffers(struct rxm_ep *rxm_ep)
 	}
 }
 
+static int rxm_cq_reprocess_directed_recvs(struct rxm_recv_queue *recv_queue)
+{
+	struct rxm_rx_buf *rx_buf;
+	struct dlist_entry *entry, *tmp_entry;
+	struct rxm_recv_match_attr match_attr;
+	struct dlist_entry rx_buf_list;
+	struct fi_cq_err_entry err_entry = {0};
+	int ret, count = 0;
+
+	dlist_init(&rx_buf_list);
+
+	recv_queue->rxm_ep->res_fastlock_acquire(&recv_queue->lock);
```

Comment:
If you changed this to just acquire the lock, we should be able to avoid pulling in the other patches from master, which aren't fixes, but performance optimizations.

Reply:
The performance optimization patches also add rxm_ep to the recv_queue struct which gets used in other places. I wish they were done in separate patches so it's easy to pick up.

Are we fine with having this optimizations in the stable branch?

Reply:
Honestly, I'd rather not rework the locking as part of a stable update unless it's an actual fix.  I think the best option might be to squash all the commits in this series into one, then manually revert the lock changes.  Maybe some tool can help here to select which changes to discard/keep.

Reply:
Makes sense. I'll skip the lock related changes then.